apps/editing-toolkit/editing-toolkit-plugin/starter-page-templates/page-template-modal/index.js

```diff
@@ -528,10 +528,10 @@ class PageTemplateModal extends Component {
 }
 
 export const PageTemplatesPlugin = compose(
-	withSelect( ( select ) => {
+	withSelect( ( select, ownProps ) => {
 		const getMeta = () => select( 'core/editor' ).getEditedPostAttribute( 'meta' );
 		const { _starter_page_template } = getMeta();
-		const isOpen = select( 'automattic/starter-page-layouts' ).isOpen();
+		const isOpen = ownProps.isOpen || select( 'automattic/starter-page-layouts' ).isOpen();
```

Comment:
I wonder if the `isPromptedFromSidebar` prop could maybe be used here. Maybe it makes it clearer that there's two modes the toggling behaviour works in, and which mode is used depends on whether it's from the sidebar or now?

```suggestion
		const isOpen = ownProps.isPromptedFromSidebar ? ownProps.isOpen : select( 'automattic/starter-page-layouts' ).isOpen();
```

Ignore this if I've misunderstood :)

Reply:
That makes sense to me!  Its a bit more clear this way.